packages/forms-web-app/src/controllers/eligibility/listed-building.js

```diff
@@ -1,46 +1,62 @@
+const logger = require('../../lib/logger');
 const { VIEW } = require('../../lib/views');
-
-const FORM_FIELD = {
-  'is-your-appeal-about-a-listed-building': {
-    id: 'is-your-appeal-about-a-listed-building',
-    items: [
-      {
-        value: 'yes',
-        text: 'Yes',
-      },
-      {
-        value: 'no',
-        text: 'No',
-      },
-    ],
-  },
-};
-
-exports.FORM_FIELD = FORM_FIELD;
+const { validIsListedBuildingOptions } = require('../../validators/eligibility/listed-building');
+const { createOrUpdateAppeal } = require('../../lib/appeals-api-wrapper');
 
 exports.getServiceNotAvailableForListedBuildings = (req, res) => {
   res.render(VIEW.ELIGIBILITY.LISTED_OUT);
 };
 
 exports.getListedBuilding = (req, res) => {
-  res.render(VIEW.ELIGIBILITY.LISTED_BUILDING, { FORM_FIELD });
+  res.render(VIEW.ELIGIBILITY.LISTED_BUILDING, { appeal: req.session.appeal });
 };
 
-exports.postListedBuilding = (req, res) => {
+exports.postListedBuilding = async (req, res) => {
   const { body } = req;
-
   const { errors = {}, errorSummary = [] } = body;
 
+  const { appeal } = req.session;
+
+  let isAppealAboutAListedBuilding = null;
+  if (validIsListedBuildingOptions.includes(req.body['is-your-appeal-about-a-listed-building'])) {
+    isAppealAboutAListedBuilding = req.body['is-your-appeal-about-a-listed-building'] === 'yes';
+  }
+
   if (Object.keys(errors).length > 0) {
     res.render(VIEW.ELIGIBILITY.LISTED_BUILDING, {
+      appeal: {
+        ...appeal,
+        eligibility: {
+          ...appeal.eligibility,
+          isListedBuilding: isAppealAboutAListedBuilding,
+        },
+      },
       errors,
       errorSummary,
-      FORM_FIELD,
     });
     return;
   }
 
-  if (body['is-your-appeal-about-a-listed-building'] === 'yes') {
+  try {
+    req.session.appeal = await createOrUpdateAppeal({
+      ...appeal,
+      eligibility: {
+        ...appeal.eligibility,
```

Comment:
Does this duplicate code entered in line 42? Not duplicating is probably more inconvenient...

Reply:
Line 44 is required or the existing key / values inside `eligibility` would be lost.

Perhaps better illustrated with a screenshot:

![Screenshot from 2021-04-12 08-56-28](https://user-images.githubusercontent.com/6329541/114360497-fe4a7d80-9b6c-11eb-8917-953c695dd9b7.png)

`a` is the original object

`x` is a wholesale copy

`y` is a copy that redefines its own `b`, but in doing so we lose the contents of `a.b`.

`p` demonstrates the additional spread (like line 44) which retains everything that exists in `a.b` and adds in new stuff, too.